gui/gui.go

```diff
@@ -122,35 +108,49 @@ type poolStats struct {
 	PoolHashRate      string
 }
 
+// headerData contains all of the necessary information to render the
+// header template.
+type headerData struct {
+	CSRF        template.HTML
+	Designation string
+	ShowMenu    bool
+}
+
 // route configures the http router of the user interface.
 func (ui *GUI) route() {
 	ui.router = mux.NewRouter()
-	ui.router.Use(csrf.Protect(ui.cfg.CSRFSecret, csrf.Secure(true)))
-
-	cssDir := http.Dir(filepath.Join(ui.cfg.GUIDir, "assets/public/css"))
-	ui.router.PathPrefix("/css/").Handler(http.StripPrefix("/css/",
-		http.FileServer(cssDir)))
-
-	imagesDir := http.Dir(filepath.Join(ui.cfg.GUIDir, "assets/public/images"))
-	ui.router.PathPrefix("/images/").Handler(http.StripPrefix("/images/",
-		http.FileServer(imagesDir)))
 
-	jsDir := http.Dir(filepath.Join(ui.cfg.GUIDir, "assets/public/js"))
-	ui.router.PathPrefix("/js/").Handler(http.StripPrefix("/js/",
-		http.FileServer(jsDir)))
-
-	ui.router.HandleFunc("/", ui.Homepage).Methods("GET")
-	ui.router.HandleFunc("/account", ui.IsPoolAccount).Methods("GET")
-	ui.router.HandleFunc("/admin", ui.AdminPage).Methods("GET")
-	ui.router.HandleFunc("/admin", ui.AdminLogin).Methods("POST")
-	ui.router.HandleFunc("/backup", ui.DownloadDatabaseBackup).Methods("POST")
-	ui.router.HandleFunc("/logout", ui.AdminLogout).Methods("POST")
-
-	// Paginated endpoints allow the GUI to request pages of data
-	ui.router.HandleFunc("/mined_blocks", ui.PaginatedMinedBlocks).Methods("GET")
-
-	// Websocket endpoint allows the GUI to receive updated values
-	ui.router.HandleFunc("/ws", ui.registerWebSocket).Methods("GET")
+	// Use a separate router without rate limiting (or other restrictions) for
+	// static assets.
+	assetsRouter := ui.router.PathPrefix("/assets").Subrouter()
+
+	assetsDir := http.Dir(filepath.Join(ui.cfg.GUIDir, "assets/public/"))
+	assetsRouter.PathPrefix("/").Handler(http.StripPrefix("/assets",
+		http.FileServer(assetsDir)))
+
+	// All other routes have rate limiting and CSRF protection applied.
+	guiRouter := ui.router.PathPrefix("/").Subrouter()
+
+	// sessionMiddleware must be run before rateLimitMiddleware.
+	guiRouter.Use(ui.sessionMiddleware)
+	guiRouter.Use(ui.rateLimitMiddleware)
+	guiRouter.Use(csrf.Protect(ui.cfg.CSRFSecret, csrf.Secure(true)))
+
+	guiRouter.HandleFunc("/", ui.Homepage).Methods("GET")
+	guiRouter.HandleFunc("/account", ui.Account).Methods("GET")
+	guiRouter.HandleFunc("/account_exist", ui.IsPoolAccount).Methods("GET")
```

Comment:
Perhaps this route could be `/account/exist`, `/poolaccount`, `/isaccount` or `/exist`? These options conform to the route definitions structure already in place, `account_exist`, with that underscore does not.

Reply:
Good catch. Reading further on this topic I believe a more proper solution would be to use the same `/account` URL as the page itself, but change the request type to `HEAD`.

https://stackoverflow.com/questions/21087913/proper-route-for-checking-resource-existence-in-a-restful-api
also links through to https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html (section 9.4 is useful)